monai/data/writers/niftiwriter.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2020 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import nibabel as nib
+
+
+def write_nifti(data, affine, file_name, revert_canonical, dtype="float32"):
+    """Write numpy data into nifti files to disk.
+
+    Args:
+        data (numpy.ndarray): input data to write to file.
+        affine (numpy.ndarray): affine information for the data.
+        file_name (string): expected file name that saved on disk.
+        revert_canonical (bool): whether to revert canonical.
+        dtype (np.dtype, optional): convert the loaded image to this data type.
+
+    """
+    assert isinstance(data, np.ndarray), 'input data must be numpy array.'
+    if affine is None:
+        affine = np.eye(4)
+
+    if revert_canonical:
+        codes = nib.orientations.axcodes2ornt(nib.orientations.aff2axcodes(np.linalg.inv(affine)))
```

Comment:
Does `np.linalg.inv(affine)` actually give up the orientation we expect of the original? If the original's axis codes were LAS they would become RAS when converted to canonical, but doesn't this line turn RAS into LPI?

Reply:
Hi @wyli ,

As you are expert at Nifti format, could you please help take a look at @ericspod 's question?
Thanks.

Reply:
@ericspod  #74 fixes the canonical reversion logic. We need to store the original affine matrix, use it to get orientation and apply the orientation to revert data.